### DISCUSSION TITLE
Economic Modeling

### DISCUSSION THREAD

**Discussion by ManuelLopezMe** [mesa/mesa#1061] [Q&A]
Hi, I'm interested in modeling a two-sector economy with a single good. For I'm very new to ABM modeling (but not python) so I was wondering how can I script two things: a trade function between different classes of agents (in this case _firms_ and _households_) and secondly, a production function that takes in inputs to generate an output (in this case _household's labor_ + resources/capital/etc => _good_). Something that could help point me in the right direction would be most helpful, thanks!

**Comment by rht** [comment]
There is a library called abcEconomics (https://github.com/ab-ce/abce), with an example for a two-sector economy: https://github.com/AB-CE/examples/tree/master/examples/2sectors. This is how the production function is generated: https://github.com/AB-CE/examples/blob/master/examples/2sectors/firm.py#L28 (you can check the function definition of `create_cobb_douglas`).
